File: 119.杨辉三角-ii.go
/*
 * @lc app=leetcode.cn id=119 lang=golang
 *
 * [119] 杨辉三角 II
 */

// @lc code=start
func getRow(rowIndex int) (currentRow []int) {
	preRow := []int{}
	// 由于只和上一层有关，所以实际上一需要维护当前行和上一行
	for i := 0; i < rowIndex+1; i++ {
		currentRow = []int{}
		for j := 0; j < i+1; j++ {
			if j == 0 || j == i {
				currentRow = append(currentRow, 1)
			} else {
				currentRow = append(currentRow, preRow[j]+preRow[j-1])
			}
		}
		// if i == rowIndex { // 索引符合条件即返回当前行
		// 	return currentRow
		// } else { // 不符合条件则重置前一行
		// 	preRow = currentRow
		// }
		preRow = currentRow
	}
	return preRow
}

// @lc code=end

File: 119.杨辉三角-ii.py
#
# @lc app=leetcode.cn id=119 lang=python3
#
# [119] 杨辉三角 II
#

# @lc code=start
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        pre_row = []
        # 维护一个前一行和当前行，节省一点空间
        for i in range(rowIndex + 1):
            current_row = []
            for j in range(i + 1):
                if j == 0 or j == i:
                    current_row.append(1)
                else:
                    current_row.append(pre_row[j]+ pre_row[j - 1])
            pre_row = current_row
        return pre_row
# @lc code=end

